chainerx_cc/chainerx/routines/creation.cc

```diff
@@ -337,4 +338,66 @@ Array Linspace(
     return out;
 }
 
+Array Tri(int64_t n, absl::optional<int64_t> m, absl::optional<int64_t> k, absl::optional<Dtype> dtype, Device& device) {
+    if (!m.has_value()) {
+        m = n;
+    }
+    if (!k.has_value()) {
+        k = 0;
+    }
+    if (!dtype.has_value()) {
+        dtype = Dtype::kFloat64;
```

Comment:
Could you change this to fp32 as well? Sorry I now noticed that e.g. eye behaves similar to your code but I think we should change it to fp32.

Reply:
For instance, `chx.arange` with floats defaults to fp32 while it for NumPy defaults to fp64. So the current routines are a bit inconsistent.

Reply:
Now the default dtype for `Tri` is fp32.